designs/tripleo/osc-tripleo.md

```diff
@@ -0,0 +1,137 @@
+
+## Background
+
+[TripleO](#tripleo-home) is the friendly name for “OpenStack on OpenStack”. It is an official OpenStack project with the goal of allowing you to deploy and manage a production cloud onto bare metal hardware using a subset of existing OpenStack components.  The deployed production cloud is called **overcloud** and the infrastructure services used for deployment are called **undercloud** as shown in the picture below. 
+
+![](./images/tripleo-services.png)  
+*TripleO Undercloud & Overcloud*  
+
+The intent of this work is to allow OSC to be more easily deployed and configured in a OpenStack cloud by using TripleO. OSC will be deployed in the **controller** node of the **overcloud** as shown below:  
+![](./images/osc-tripleo-services.png)  
+*OSC In the TripleO Overcloud*  
+
+## Constraints and Assumptions
+### OSC Packaged Binaries
+TripleO can only deploy OSC through a template if the OSC binaries are available in a public repository. There are multiple paths to achieve this:  
+1.  RPM package.
+2.  Docker container image on an arbitrary Docker repository.
+3.  Docker container image on a Kolla repository.
+
+The source of the OSC binaries is outside of the scope for this document but it assumes option *3. Kolla repository* at the moment. For any other choice the OSC THT may need changes and/or additional Puppet manifests might be needed.   
+
+### Puppet TripleO
+No changes or new files will be needed in the Puppet TripleO Repository: [https://github.com/openstack/puppet-tripleo](https://github.com/openstack/puppet-tripleo)   
+
+## Design Changes
+A THT describes the OpenStack deployment in Heat Orchestration Template YAML files and Puppet manifests. A template for the OSC service will be created and registered with TripleO. The goal is for all the changes described on this section to be upstreamed to the [THT repository](#tripleo-repo)
+
+### OSC THT  
+
+#### Template Location  
+The OSC THT will be  `/tripleo-heat-templates/docker/services/osc.yaml`. The resource must be registered in the environment file `tripleo-heat-templates/environments/docker.yaml` as follows:  
+```OS::TripleO::Services::OSC: ../docker/services/osc.yaml```  
+
+With these files in place the user can include the OSC service as a resource in the *overcloud* by adding `OS::TripleO::Services::OSC` to the *controller* role during deployment and providing *--environment* files with the values of the parameters defined on the template.   
+
+#### Template Content
+The file `../docker/services/osc.yaml` starts with the template version and a description, followed a set of parameters, a list of resources and outputs generated by the template. For details on THTs, see the [guide](#tht-guide).  
+
+```yaml  
+heat_template_version: rocky
+
+description: >
+  Open Security Controller: service for orchestration of virtual network security functions.
+
+ parameters:
+  #  The paramater EndpointMap is required for all service templates*.
+  EndpointMap:
```

Comment:
Not clear how the EndpointMap contains and how is this used? I tried to google, but not much information is available.
could you add an example? and give description about this?

Reply:
It is not fully clear to me either. However I did put a reference to the document that says it is required "### [THT Guide](https://docs.openstack.org/heat/pike/template_guide/hot_guide.html) " . We can clarify that once we review with Luke. I am tracking another user story to update this document after the first review with RH and will ensure to add that detail. Adding a TODO now.